### ISSUE TITLE
Parameter min and max values are not properly scaled 

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#1878]
When changing the min and max value of a parameter, the scale factor is not to applied leading to an incorrect behavior during the fit process.
For example, following the model definition analysis_3d:
```
spatial_model = SkyPointSource(lon_0="0.11 deg", lat_0="-0.41 deg")
spectral_model = PowerLaw(
    index=2.9, amplitude="1e-10 cm-2 s-1 TeV-1", reference="1 TeV"
)
model = SkyModel(spatial_model=spatial_model, spectral_model=spectral_model)
```
If you now want to set the minimum value of `spectral_model.amplitude` to 1e-18  cm-2 s-1 TeV-1, you have to use:
```
spectral_model.parameters['amplitude'].min=-18
```
because scaling is applied by default. If you simply do:
```
spectral_model.parameters['amplitude'].min=1e-18
```
the amplitude will be limited at 1e(1e-18) during the fit...

**Comment by cdeil** [comment]
@registerrier - Yes. We just didn't have time to work on limits before the 0.8 release.
If you make changes here, please add a unit test to establish behaviour.

In principle this is a simple addition, although when it comes to the question how fitting backends handle limits, it becomes complicated.

MINUIT does a parameter transform when min / max are set, Sherpa I'm not sure, I think they don't but have somehow a two-stage limit system. Scipy has nothing, so before calling scipy.optimise we should do it. Overall probably we should do limit handling and not do it in the optimiser interface. That's e.g. what this guy did before calling into scipy: http://lmfit.github.io/lmfit-py/bounds.html#bounds-chapter

I think @adonath wanted to take on this task, but of course if you want to go ahead.

But somehow fix the broken notebook first, no? (e.g. by increasing the exposure by factor 10 or whatever), this is a few-day coding + review task.

**Comment by registerrier** [comment]
OK. I will provide a correction to the notebook by adding min values to the amplitude parameter in issue #1868. This will have to be updated once we define a better implementation for `Parameter.min` and `Parameter.max`

**Comment by aionviana** [comment]
Hi there. I also found this bug when trying to fix min and max values for the position of a source. This is particularly important for very complex regions with multiple sources. However, it seems that even fixing the scale beforehand, a re-scale is done at the fitting procedure and it is not propagated to the min and max values. 

For instance, take this model for a 3d analysis:

```
model_input = SkyModel(
    SkyPointSource("0.01d", "0.01d"),
    ExponentialCutoffPowerLaw(amplitude="1e-12 cm-2 s-1 TeV-1", index=1.0, reference="1 TeV", lambda_=1 / u.TeV)
)
```
The actual latitude position of the source is around lat = -0.04 deg, so we can fix min, max and the scale to help the fitting procedure:

```
model_input.parameters["lat_0"].scale = 1
model_input.parameters["lat_0"].min = -0.2 
model_input.parameters["lat_0"].max = 0.2
```
So the parameter latitude is defined as:

```
>>> model_input.parameters["lat_0"] 
Parameter(name='lat_0', value=0.01, factor=0.01, scale=1, unit='deg', min=-0.2, max=0.2, frozen=False)
```
Then we run the fit procedure
```
fit = MapFit(
    model=model_input,
    counts=cmaps["counts"],
    exposure=cmaps["exposure"],
    background=cmaps["background"],
    psf=psf_kernel,
    edisp=edisp,
)
result = fit.run(optimize_opts={"print_level": 1})


+ | Name | Value | Hesse Error | Minos Error- | Minos Error+ | Limit- | Limit+ | Fixed?
-- | -- | -- | -- | -- | -- | -- | -- | --
0 | par_000_lon_0 | -5.48948 | 0.0120675 |   |   |   |   | No
1 | par_001_lat_0 | -0.2 | 1.77816e-05 |   |   | -0.2 | 0.2 | No
2 | par_002_index | 2.17617 | 0.00531099 |   |   |   |   | No
3 | par_003_amplitude | 2.92384 | 0.0128765 |   |   |   |   | No
4 | par_004_reference | 1 | 1 |   |   |   |   | Yes
5 | par_005_lambda_ | 0.0859259 | 0.00196024 |   |   |   |   | No
```
The fit converges properly, but the parameter latitude is found at the minimum of -0.2 due to a change of scale from 1 to 0.01. So instead of having a minimum at -0.2deg as we wanted, it is now at -0.002 deg:
 ```
>>> result.model.parameters["lat_0"] 
Parameter(name='lat_0', value=-0.001999999999994223, factor=-0.1999999999994223, scale=0.01, unit='deg', min=-0.2, max=0.2, frozen=False)
```

**Comment by cdeil** [comment]
Discussed with @adonath and @registerrier . I'll take care of this for v0.9:

- This only requires small changes in the optimiser interfaces
- `min` / `max` attributes should correspond to `value`, we don't need extra Parameter attributes.
- In `optimize_iminuit`, adjust the code so that `limit_` corresponds to `par.value / par.factor`.
- In `optimize_sherpa`, pass factor and min / max adjusted with factors as well
- Main work is to review and improve existing tests.

**Comment by cdeil** [comment]
Fixed in #1904 .

I will follow up with a PR with more tests for `Fit` and asserts on parameter errors later today to see if scales are used correctly with the covariance matrix.

@registerrier - I don't see any min or max any more in `analysis_3d.ipynb` If you think it's useful there or anywhere else, please add it back and see if it now behaves as it should.
